collector/mongod/index_usage.go

```diff
@@ -50,91 +50,84 @@ func (indexStats *IndexStatsList) Export(ch chan<- prometheus.Metric) {
 		if err != nil {
-			_, logSFound := logSuppressIS[dbName]
-			if !logSFound {
-				log.Errorf("%s. Index usage stats will not be collected for this db. This log message will be suppressed from now.", err)
-				logSuppressIS[dbName] = true
+			if !logSuppressIS.Contains(dbName) {
+				log.Warnf("%s. Index usage stats will not be collected for this db. This log message will be suppressed from now.", err)
+				logSuppressIS.Add(dbName)
 			}
-		} else {
+			continue
+		}
 
-			type collListItem struct {
-				Name string `bson:"name,omitempty"`
-				Type string `bson:"type,omitempty"`
+		logSuppressIS.Delete(dbName)
+		for _, collName := range collNames {
+			if common.IsSystemCollection(collName) {
+				continue
+			}
+
+			fullCollName := common.CollFullName(dbName, collName)
+			collIndexUsageStats := IndexStatsList{}
+			c, err := client.Database(dbName).Collection(collName).Aggregate(context.TODO(), []bson.M{{"$indexStats": bson.M{}}})
+			if err != nil {
+				if !logSuppressIS.Contains(fullCollName) {
+					log.Warnf("%s. Index usage stats will not be collected for this collection. This log message will be suppressed from now.", err)
+					logSuppressIS.Add(fullCollName)
+				}
+				continue
 			}
 
-			delete(logSuppressIS, dbName)
 			for c.Next(context.TODO()) {
-				coll := &collListItem{}
-				err := c.Decode(&coll)
-				if err != nil {
+				s := &IndexUsageStats{}
+				if err := c.Decode(s); err != nil {
 					log.Error(err)
 					continue
 				}
+				collIndexUsageStats.Items = append(collIndexUsageStats.Items, *s)
+			}
 
-				collIndexUsageStats := IndexStatsList{}
-				c, err := client.Database(dbName).Collection(coll.Name).Aggregate(context.TODO(), []bson.M{{"$indexStats": bson.M{}}})
-				if err != nil {
-					_, logSFound := logSuppressIS[dbName+"."+coll.Name]
-					if !logSFound {
-						log.Errorf("%s. Index usage stats will not be collected for this collection. This log message will be suppressed from now.", err)
-						logSuppressIS[dbName+"."+coll.Name] = true
-					}
-				} else {
-
-					for c.Next(context.TODO()) {
-						s := &IndexUsageStats{}
-						if err := c.Decode(s); err != nil {
-							log.Error(err)
-							continue
-						}
-						collIndexUsageStats.Items = append(collIndexUsageStats.Items, *s)
-					}
-
-					if err := c.Err(); err != nil {
-						log.Error(err)
-					}
-
-					if err := c.Close(context.TODO()); err != nil {
-						log.Errorf("Could not close Aggregate() cursor, reason: %v", err)
-					}
-
-					delete(logSuppressIS, dbName+"."+coll.Name)
-					// Label index stats with corresponding db.collection
-					for i := 0; i < len(collIndexUsageStats.Items); i++ {
-						collIndexUsageStats.Items[i].Database = dbName
-						collIndexUsageStats.Items[i].Collection = coll.Name
-					}
-					indexUsageStatsList.Items = append(indexUsageStatsList.Items, collIndexUsageStats.Items...)
-				}
+			if err := c.Err(); err != nil {
```

Comment:
Do we need the assignment?

Reply:
Nope, I assume it's just code style from `mongo-go-driver` example. Should I change it? (same code all over the project)

Reply:
We need to stop somewhere with this PR :)

Reply:
Ok, next time)